cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTable.java

```diff
@@ -73,9 +75,10 @@
   public HBaseTable(String name, ConflictDetection level, Configuration hConf, boolean enableReadlessIncrements)
     throws IOException {
     super(name, level, enableReadlessIncrements);
-
     hTableName = HBaseTableUtil.getHBaseTableName(name);
-    HTable hTable = new HTable(hConf, hTableName);
+    HBaseTableUtil tableUtil = new HBaseTableUtilFactory().get();
+    TableId tableId = TableId.from(hTableName);
```

Comment:
this a weird flow, we get hTableName from ```HBaseTableUtil.getHBaseTableName(name)``` then create a TableId that we pass back into tableUtil.  What about just passing in TableId in the constructor?